developer_manual/index.rst

```diff
@@ -10,4 +10,5 @@ Here you will find the documentation for developers.
    :maxdepth: 1
 
    guide-api
-   api
\ No newline at end of file
+   api
+   contributing-guide
```

Comment:
This isn't a contribuding guide. The contributing guide is this:

https://github.com/LibreSign/libresign/blob/main/CONTRIBUTING.md

And we will need to move parts of CONTRIBUTING to documentation site.

Would be good to split this file into dedicated files to don't create the entire documentation into a single file.

I suggested here a tree of contents:

https://github.com/LibreSign/documentation/pull/49#discussion_r2150371780